src/functions/def.js

```diff
@@ -4,37 +4,86 @@ import ParseError from "../ParseError";
 import {assertNodeType} from "../parseNode";
 
 const globalMap = {
+    "\\global": "\\global",
+    "\\long": "\\\\globallong",
+    "\\\\globallong": "\\\\globallong",
     "\\def": "\\gdef",
     "\\gdef": "\\gdef",
+    "\\edef": "\\xdef",
+    "\\xdef": "\\xdef",
+    "\\let": "\\\\globallet",
+    "\\futurelet": "\\\\globalfuture",
 };
 
-// Basic support for macro definitions:
-//     \def\macro{expansion}
-//     \def\macro#1{expansion}
-//     \def\macro#1#2{expansion}
-//     \def\macro#1#2#3#4#5#6#7#8#9{expansion}
-// Also the \gdef and \global\def equivalents
+const checkControlSequence = (tok) => {
+    const name = tok.text;
+    if (/^(?:[\\{}$&#^_]|EOF)$/.test(name)) {
+        throw new ParseError("Expected a control sequence", tok);
+    }
+    return name;
+};
+
+const getRHS = (parser) => {
+    let tok = parser.gullet.popToken();
+    if (tok.text === "=") { // consume optional equals
+        tok = parser.gullet.popToken();
+        if (tok.text === " ") { // consume one optional space
+            tok = parser.gullet.popToken();
+        }
+    }
+    return tok;
+};
+
+const letCommand = (parser, name, tok, global) => {
+    let macro = parser.gullet.macros.get(tok.text);
+    if (macro == null) {
+    // if macro is undefined at this moment, set noexpand to 2
+    // and unexpandable to not expand it later and pass to the parser
```

Comment:
Is this to handle a case like `\let\foo=\bar \def\bar{hello, world} \foo` where the value of the `let` is defined after the `let`?

Reply:
This is to handle cases where non-macros are overwritten after `\let`, e.g., `\let\foo=\frac \def\frac{123} \foo12`.